calicoctl/commands/test_ipam.go

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2019 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package commands
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/docopt/docopt-go"
+
+	"github.com/projectcalico/calicoctl/calicoctl/commands/constants"
+	"github.com/projectcalico/calicoctl/calicoctl/commands/test_ipam"
+)
+
+func TestIPAM(args []string) error {
```

Comment:
This is neat and seems useful to have.

however, I worry about adding it to calicoctl itself since users might not really understand it.


I think at a minimum we should have it be under some clearly named subcommand - e.g., `calicoctl unsupported test-ipam` or something like that.

Even better might just be to have a second tool that this repo builds that is only used for test purposes and not actually supplied to users.

And even better, I think my dream would be that instead of building a tool at all, we simply write the tests that require this in golang so they can just import the logic they need. WDYT?